### PR TITLE
(DOCSP-29899): Ready site for MVP publish

### PR DESCRIPTION
**Pull request by mongodben**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-29899
Staging - https://docs-mongodbcom-staging.corp.mongodb.com/kotlin/docsworker-xlarge/DOCSP-29899/

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Did you run a spell-check?
- [ ] Did you run a grammar-check?
- [ ] Are all the links working?

### CODE REVIEW

File: source/index.txt [hunk 1]
```diff
@@ -27,7 +27,7 @@ MongoDB driver for server-side Kotlin applications using coroutines. Download it
 following our Quick Start guide.
 
 If your Kotlin application requires synchronous processing, use the
-:driver:`Sync Driver <TODO:(DOCSP-29157)>` which uses synchronous operations
+:driver:`Sync Driver <{+api-kotlin+}/apidocs/mongodb-driver-kotlin-sync/>` which uses synchronous operations
```
Comment (cbullinger) [inline source/index.txt:30]:
```suggestion
`Sync Driver <{+api-kotlin+}/apidocs/mongodb-driver-kotlin-sync/>` which uses synchronous operations
```
this 404s (`:driver:` prefixes the link with extraneous `https://www.mongodb.com/docs/drivers`)

Comment (mongodben) [reply source/index.txt:30]:
```suggestion
`Sync Driver <{+api-kotlin+}/apidocs/mongodb-driver-kotlin-sync/>`__ which uses synchronous operations
```

File: source/fundamentals/connection/connection-options.txt [hunk 2]
```diff
@@ -192,10 +192,15 @@ parameters of the connection URI to specify the behavior of the client.
...
-     - Specifies the :doc:`authentication mechanism
-       </fundamentals/auth>` that the driver should use if a credential
+     - Specifies the authentication mechanism
+       that the driver should use if a credential
        was supplied.
 
+.. TODO:(DOCSP-29261) refactor above to:
+..  Specifies the :doc:`authentication mechanism
+..  </fundamentals/auth>` that the driver should use if a credential
+..  was supplied.
```
Comment (ccho-mongodb) [inline source/fundamentals/connection/connection-options.txt:202]: Question:
I'm not sure I understand what the work is in the referenced ticket or in this comment. Could someone clarify in the ticket? And once the ticket is updated, is this comment still necessary?

File: source/fundamentals/connection/mongoclientsettings.txt [hunk 3]
```diff
@@ -84,7 +84,7 @@ connection behavior:
        :ref:`TLS/SSL settings <mcs-ssl-settings>`.
 
    * - ``autoEncryptionSettings()``
-     - | Sets the :ref:`auto-encryption settings <auto-encryption-decryption-kotlin>`.
+     - | Sets the auto-encryption settings.
```
Comment (ccho-mongodb) [inline source/fundamentals/connection/mongoclientsettings.txt:87]: Suggestion:
I think the term should be "automatic encryption".



```suggestion
     - | Sets the automatic encryption options. To learn more about in-use automatic encryption, see the :manual:`Client-Side Field Level Encryption </core/csfle>` and :manual:`Queryable Encryption </core/queryable-encryption/>` guides in the Server manual.
```

**Comment by ashni-mongodb** [comment]
Hey! Looks like some of the Quick Start text refers to Java -- the steps around Connecting Your Application in particular. Screenshots also refer to Java but until CLOUD adds Kotlin to the drop-down list I'm not sure if we'll be able to pull new screenshots.

**Comment by ccho-mongodb** [comment]
In source/fundamentals/connection/connect.txt (or whatever other included files it may reference), the connection example shows:

```
val uri = CONNECTION_URI_PLACEHOLDER
```

Suggestion:
Use the same pattern as the rest of the drivers: "<connection string>" if possible.

**Comment by ashni-mongodb** [comment]
I think the "Core" link on [this page](https://docs-mongodbcom-staging.corp.mongodb.com/kotlin/docsworker-xlarge/DOCSP-29899/api-documentation/) is pointing to 4.10 while the rest are pointing to 4.9!

**Comment by ashni-mongodb** [comment]
In the "Issues and Help" section, specifically the "Pull Requests" subsection that includes info on source checkout / starting a branch -- we might want to route folks further into the Kotlin driver instead of just the overall mongo-java-driver. What do y'all think?

File: source/fundamentals/connection/connection-options.txt (hunk 2 above)
Comment (mongodben) [reply source/fundamentals/connection/connection-options.txt:202]: ticket/comment refer to creating to the page `/fundamentals/auth` and then also linking to it here.

File: source/fundamentals/connection/mongoclientsettings.txt [hunk 4]
```diff
@@ -96,18 +96,28 @@ connection behavior:
...
+.. TODO(DOCSP-29268) set sentence above to
+.. Sets the :ref:`compressors <compression>` to use for compressing
+.. messages to the server.
+
    * - ``credential()``
      - Sets the :ref:`credential <authentication-mechanisms-kotlin>`.
 
+.. TODO:(DOCSP-29261) set sentence above to
+.. Sets the :ref:`credential <authentication-mechanisms-kotlin>`.
+
```
[the remark this answers is not in the document]
Comment (mongodben) [reply source/fundamentals/connection/mongoclientsettings.txt:120]: good point. i wasn't thinking too hard about these TODOs will be completed in the next couple weeks. but linking to reference docs seems like a good tmp fix. will add

**Comment by ccho-mongodb** [comment]
Question:
In the Quick Start instructions for Atlas cluster setup, will there be a Kotlin driver option in the Driver select box? The current text and screenshots refer to Java, and not what's currently there. Maybe Kotlin doesn't exist, but the current version of this refers to "Java" and "4.9 or Later" which doesn't exist yet.

**Comment by mongodben** [comment]
> Question: In the Quick Start instructions for Atlas cluster setup, will there be a Kotlin driver option in the Driver select box? The current text and screenshots refer to Java, and not what's currently there. Maybe Kotlin doesn't exist, but the current version of this refers to "Java" and "4.9 or Later" which doesn't exist yet.

ashni commented smthn similar. will update to current java version and add an admonition explaining that you can use the java connection uri until kotlin is added to ui. 

not sure exactly when kotlin will be added, but supposed to happen before .local NYC i believe.

**Comment by ccho-mongodb** [comment]
Suggestion:
I found the CRUD Operations > Write Operations > Bulk Operations code examples a little difficult to follow because of the use of numbers in both variable name and constructor values. I wonder if using some other name for the variables that does not include numbers might work better. For example:

"doc5" -> "apples"
"doc6" -> "oranges"

**Comment by ccho-mongodb** [comment]
A few suggestions on the Issues & Help page:
- I think the information about the "[JIRA] bug reports" should be an admonition rather than one that includes bold text since we usually reserve that for terminology.
- I think gradle doesn't need to be in bold. Instead, if there needs to be formatting, monospace might work best as the style guide recommends it for "command names".

Question:
Will there be a separate KOTLIN driver project? I saw one for Kotlin BSON, so just curious.

**Comment by mongodben** [comment]
> Suggestion: I found the CRUD Operations > Write Operations > Bulk Operations code examples a little difficult to follow because of the use of numbers in both variable name and constructor values. I wonder if using some other name for the variables that does not include numbers might work better. For example:
> 
> "doc5" -> "apples" "doc6" -> "oranges"

for this one, i agree with you that the examples could be improved. however, here the examples are following the pattern of the [same page in the java driver docs](https://www.mongodb.com/docs/drivers/java/sync/current/fundamentals/crud/write-operations/bulk/), which we're trying to do for the initial draft of these docs.

going to keep this the same for now, though have created a ticket to improve this example down the line

**Comment by mongodben** [comment]
> A few suggestions on the Issues & Help page:
> 
> * I think the information about the "[JIRA] bug reports" should be an admonition rather than one that includes bold text since we usually reserve that for terminology.
> * I think gradle doesn't need to be in bold. Instead, if there needs to be formatting, monospace might work best as the style guide recommends it for "command names".
> 
> Question: Will there be a separate KOTLIN driver project? I saw one for Kotlin BSON, so just curious.

i didn't see in jira. but not sure if there will be down the line. @jyemin or @ashni-mongodb do you know?

**Comment by jyemin** [comment]
No, we'll be using the Kotlin component in the JAVA project

**Review by cbullinger — APPROVED** [review]
A couple small things, but other updates all look good:
- modify opening sentence on [Builders page ](https://docs-mongodbcom-staging.corp.mongodb.com/kotlin/docsworker-xlarge/DOCSP-29899/fundamentals/builders/) to not refer to the unpublished Builders guides
- broken links in API Reference toc section -- the BSON kotlinx.serialization and Core links 404 when you click from the toc (they work fine from the [API Documentation landing page](https://docs-mongodbcom-staging.corp.mongodb.com/kotlin/docsworker-xlarge/DOCSP-29899/api-documentation/))
- and fyi -- selecting a version from the dropdown breaks the staging link

**Comment by mongodben** [comment]
> A couple small things, but other updates all look good:
> 
> * modify opening sentence on [Builders page ](https://docs-mongodbcom-staging.corp.mongodb.com/kotlin/docsworker-xlarge/DOCSP-29899/fundamentals/builders/) to not refer to the unpublished Builders guides

changin

> * broken links in API Reference toc section -- the BSON kotlinx.serialization and Core links 404 when you click from the toc (they work fine from the [API Documentation landing page](https://docs-mongodbcom-staging.corp.mongodb.com/kotlin/docsworker-xlarge/DOCSP-29899/api-documentation/))

fixin

> * and fyi -- selecting a version from the dropdown breaks the staging link

this is expected and not an issue to the best of my understanding

**Pull request merged**